### PR TITLE
Fix huge amount of link names in published state

### PR DESCRIPTION
**Pull request by ziyan**
GIL taken too long by:

```
Thread 24 (Thread 0x7f7aca7fc700 (LWP 26409)):
Traceback (most recent call first):
  File "/opt/lib/python2.7/site-packages/openravepy/_openravepy_/interfaces/TaskManipulation.py", line 239, in UnchuckFingers
    res = self.prob.SendCommand(cmd)
  File "/opt/lib/python2.7/site-packages/mujinplanningcommon/planning/realtimerobottask3.py", line 4794, in _UnchuckGripper
    finalfingers, unchuckFingersTraj = self.taskmanip.UnchuckFingers(translationstepmult=0.01, execute=False, outputtrajobj=False, outputfinal=True, movingdir=[-value for value in chuckingdirection])
  File "/opt/lib/python2.7/site-packages/mujinplanningcommon/planning/itlplanning3/itlexecutor.py", line 691, in _ExecuteUnchuckGripper
    asyncExec=command['doAsync'],
  File "/opt/lib/python2.7/site-packages/mujinplanningcommon/planning/itlplanning3/itlexecutor.py", line 1258, in _QueueOrExecuteITLCommand
    response, traj = self._ExecuteUnchuckGripper(executeparameters, command, releaseTargetData=releaseTargetData)
  File "/opt/lib/python2.7/site-packages/mujinplanningcommon/planning/itlplanning3/itlexecutor.py", line 530, in _RunExecuteThread
    response = self._QueueOrExecuteITLCommand(executeparameters)
  File "/opt/lib/python2.7/site-packages/mujincommon/__init__.py", line 245, in <lambda>
    return lambda *args, **kwargs: methodref()(*args, **kwargs)
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 774, in __bootstrap
    self.__bootstrap_inner()
```
    
```
#0  boost::shared_ptr<OpenRAVE::KinBody::Link const>::shared_ptr (r=<synthetic pointer>..., this=0x7f7aca7f7b80) at /usr/include/boost/smart_ptr/shared_ptr.hpp:405
#1  OpenRAVE::KinBody::CheckSelfCollision (this=0x7f7abc07b8b0, report=..., collisionchecker=...) at ../../../../../checkoutroot/openrave/src/libopenrave/kinbodycollision.cpp:98
#2  0x00007f7b0eb0f1ed in GrasperPlanner::_CheckCollision (this=0x7f7ac0562390, plink=..., targetbody=...) at ../../../../../checkoutroot/openrave/plugins/grasper/grasperplanner.cpp:590
#3  0x00007f7b0eb0e6d6 in GrasperPlanner::_CheckCollision (this=0x7f7ac0562390, pjoint=..., targetbody=...) at ../../../../../checkoutroot/openrave/plugins/grasper/grasperplanner.cpp:549
#4  0x00007f7b0eb163f9 in GrasperPlanner::PlanPath (this=0x7f7ac0562390, ptraj=..., planningoptions=<optimized out>) at ../../../../../checkoutroot/openrave/plugins/grasper/grasperplanner.cpp:471
#5  0x00007f7b0bfeb38b in TaskManipulation::UnchuckFingers (this=0x7f7ac053df10, sout=..., sinput=...) at ../../../../../checkoutroot/openrave/plugins/rmanipulation/taskmanipulation.cpp:1332
#6  0x00007f7b4743914b in boost::function2<bool, std::ostream&, std::istream&>::operator() (a1=..., a0=..., this=0x7f7ac053f030) at /usr/include/boost/function/function_template.hpp:771
#7  OpenRAVE::InterfaceBase::SendCommand (this=this@entry=0x7f7ac053df10, sout=..., sinput=...) at ../../../../../checkoutroot/openrave/src/libopenrave/interface.cpp:115
#8  0x00007f7b0bfd4cfa in TaskManipulation::SendCommand (this=0x7f7ac053df10, sout=..., sinput=...) at ../../../../../checkoutroot/openrave/plugins/rmanipulation/taskmanipulation.cpp:198
```

### CODE REVIEW

File: src/libopenrave/kinbodycollision.cpp [hunk 1]
```diff
@@ -85,14 +85,16 @@ bool KinBody::CheckSelfCollision(CollisionReportPtr report, CollisionCheckerBase
             RAVELOG_WARN_FORMAT("grabbed body on %s has already been destroyed, ignoring.", GetName());
             continue;
         }
+        if( !pbody->IsEnabled() ) {
```
[the remark this answers is not in the document]
Comment (Puttichai) [reply src/libopenrave/kinbodycollision.cpp:88]: @ziyan What proper fix do you have in mind for this?

**Comment by rdiankov** [comment]
@ziyan is this pull request still necessary?

**Comment by ziyan** [comment]
> @ziyan is this pull request still necessary?

Probably not. One nitpick with the current `production` branch is that it still publishes some `40 * 36` link names in the published message.

**Comment by rdiankov** [comment]
@ziyan err... that's not good, i thought we resolved that on robotbridges already... there's a fix necessary on openrave side?

**Comment by rdiankov** [comment]
@ziyan is this still necessary?

**Comment by rdiankov** [comment]
@ziyan still necessary?

**Comment by ziyan** [comment]
> > @ziyan is this pull request still necessary?
> 
> 
> 
> Probably not. One nitpick with the current `production` branch is that it still publishes some `40 * 36` link names in the published message.

This problem hasn't been solved yet.

**Comment by ziyan** [comment]
Closing since there might be a better solution.

**Pull request closed without merging**